### PR TITLE
Default to latest Go.

### PR DESCRIPTION
**Pull request by shrimpy**
Ran smoke test and all pass.

### CODE REVIEW

File: lib/templates/deploy.batch.go.template [hunk 1]
```diff
@@ -22,8 +13,23 @@ IF /I "%IN_PLACE_DEPLOYMENT%" EQU "1" (
 
 ECHO Prepare workspace
 IF "%GOROOT%" == "" (
-    :: select go version TODO
-    SET GOROOT=D:\Program Files\go\1.4.2
+  IF DEFINED GO_WEB_CONFIG_TEMPLATE (
+    :: Sort folder in reverse alphabetically order (-N), and folder only (G)
```
Comment (davidebbo) [inline lib/templates/deploy.batch.go.template]: This is probably not correct, as you'd end up with 1.4.2 > 1.4.10. Doing correct SemVer comparison can be challenging.

Comment (shrimpy) [inline lib/templates/deploy.batch.go.template]: right.. miss some test case.. my first attempt was to use another powershell script to do the SemVer comparison and output to a file.

but seems like once we spin out another process, we will lost the stdin stream. won`t able to read file anymore

e.g 
````
set /p myvar=<%pathToFIle%
````

let me dig a bit deeper

**Comment by davidebbo** [comment]
:shipit:

File: lib/templates/deploy.batch.go.template [hunk 2]
```diff
@@ -22,8 +13,26 @@ IF /I "%IN_PLACE_DEPLOYMENT%" EQU "1" (
 
 ECHO Prepare workspace
 IF "%GOROOT%" == "" (
-    :: select go version TODO
-    SET GOROOT=D:\Program Files\go\1.4.2
+  IF DEFINED GO_WEB_CONFIG_TEMPLATE (
+    CALL PowerShell -NoProfile -NoLogo -ExecutionPolicy unrestricted -Command "%SELECT_LATEST_VERSION_CMD% '%ProgramW6432%\Go' '%DEPLOYMENT_TEMP%\___GO_RUNTIME.tmp'"
```
Comment (mathewc) [inline lib/templates/deploy.batch.go.template]: SELECT_LATEST_GO_VERSION_CMD

**Comment by shrimpy** [comment]
merged in https://github.com/projectkudu/KuduScript/commit/bff38d6c715efeaca19986ebacaba77234e43539

**Pull request closed without merging**